src/components/quiz/QuizQuestions.tsx

```diff
@@ -1,18 +1,91 @@
-import React from "react";
+import React, { useState } from "react";
 import { Question } from "./questions";
-import { Reward } from "./Reward";
+import { UserDetailsForm } from "./UserDetailsForm";
+import './QuizQuestions.scss';
 
 interface QuizQuestionsProps {
-  questions: Question[];
+   questions: Question[];
+   score: number;
+   setScore: (score: number) => void;
+   isEndOfQuestion: boolean;
+   setIsEndOfQuestion: (isEndOfQuestion: boolean) => void;
 }
 
+
 export const QuizQuestions: React.FunctionComponent<QuizQuestionsProps> = ({
-  questions,
+  questions,score,setScore,isEndOfQuestion,setIsEndOfQuestion
 }) => {
+
+  const [activeQuestionIndex, setActiveQuestionIndex] = useState(0);
+  const activeQuestion = questions[activeQuestionIndex];
+  const [activeQuestionClicked, setActiveQuestionClicked] = useState(false);
+  const [message, setMessage] = useState("");
+  
+  const selectAnswer = (answerIndex: number) => {
+    setActiveQuestionClicked(true);
+    if (answerIndex === activeQuestion.correctAnswerIndex) {
+      setScore(score + 1);
+      setMessage('Correct!!!')
+    } else{
+      setMessage('Incorrect!!!')
+    } 
+
+    isQuizFinished();
+  }
+
+  const isQuizFinished = ()=>{
+    if(activeQuestionIndex === 10)
+    setIsEndOfQuestion(true);
+  }
+
+  const renderResult = () => {
+    if (activeQuestionClicked !== false) {
+      return (
+      <div className="quiz-questions">
+        <h3>{message}</h3>
+        <p>{activeQuestion.answerDetailsText}</p>
+          <button className="next-question" onClick={() => { moveToNextQuestion() }} >NextQuestion </button>         
```

Comment:
Make sure you use whitespace carefully! Spaces are not neutral
```suggestion
          <button className="next-question" onClick={moveToNextQuestion}>Next Question</button>         
```